components/net/http_loader.rs

```diff
@@ -1102,6 +1105,18 @@ fn http_network_fetch(request: &Request,
     response.headers = res.response.headers.clone();
     response.referrer = request.referrer.to_url().cloned();
 
+    {
+        let http_message = res.response.get_ref().downcast_ref::<Http11Message>().unwrap();
```

Comment:
Yes. I could more simply just check that the request's URL scheme was "https", but this checks that the very stream that handles the bits over the wire actually uses an `HttpsStream`, so I think that is better. And we will need that code when we will finally look at the actual cypher anyway.